LokiLabs/microsoft-graph-devx-api
# AB#38712 Check user resource ownership

## Overview

Using the user's passed in Graph delegated token, I retrieved the tenantId and user id from the token. The purpose of this PR is to use the token and the user id to make a graph call to check if the resource (teams or chat) that the client is trying to make a request on is owned by the user.

## Testing Instructions

NOTE: Alot of overhead is required to test this PR. to make it easier, ping me on Teams for a ClientId and ClientSecret to avoid registering an app and installing a Teams app on a test tenant with the same ClientId.

* Open the file appsettings.json
* In ClientId, paste the clientId of your app registration
* In ClientSecret, paste the client secret of your app registration 
* Head over to [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) 
* Go to Access token and copy the access token
* Run the branch locally and open Postman
* Add your access token that was copied from GE into the Authorization tab as a 'Bearer Token' 
* Make a call to the API using the endpoint `/graphproxy/{ADD YOUR GRAPH QUERY HERE}`. Example: `/v1.0/teams/{team-id}/channels`
* Make sure its a Graph call on a Teams resource (Team or Chat) that you are a owner of and have the GE Teams app (with the same client Id as the you pasted in appsettings.json) installed on (in order for the Graph request to work)

## Expected behavior
* If you have the teams app installed on the resource you are trying to call AND you are the owner of that resource, you will receive a 200 OK with the response
*  If you have the teams app installed on the resource you are trying to call AND you are the NOT owner of that resource, you will receive a 403 forbidden with an empty response
* If you dont have the teams app installed on the resource you are trying to call, you will receive a 403 forbidden with an error message from Graph.
